proxy/HostStatus.h

```diff
@@ -46,7 +49,26 @@ struct HostStatRec_t {
   unsigned int down_time; // number of seconds that the host should be down, 0 is indefinately
 };
 
-const std::string stat_prefix = "host_status.";
+struct Reasons {
+  const char *ACTIVE = "active";
+  const char *LOCAL  = "local";
+  const char *MANUAL = "manual";
+
+  const char *reasons[3] = {ACTIVE, LOCAL, MANUAL};
+
+  bool
+  validReason(const char *reason)
+  {
+    for (const char *i : reasons) {
+      if (strncmp(i, reason, strlen(reason)) == 0) {
```

Comment:
Better, but it still calls `strlen` every loop iteration even though the value never changes. There is also the issue that `strlen(reason)` could be larger than `strlen(i)`. If you're assuming `i` is null terminated, then you're also assuming `reason` is (or `strlen` would break) so why not just `strcmp`?

Reply:
@SolidWallOfCode  i'll change it to use strcmp.